src/Components/Components/src/Forms/InputComponents/InputBase.cs

```diff
@@ -0,0 +1,189 @@
+                    parsingFailed = false;
+                    CurrentValue = default;
+                }
+                else if (TryParseValueFromString(value, out var parsedValue, out var validationErrorMessage))
+                {
+                    parsingFailed = false;
+                    CurrentValue = parsedValue;
+                }
+                else
+                {
+                    parsingFailed = true;
+
+                    if (_parsingValidationMessages == null)
+                    {
+                        _parsingValidationMessages = new ValidationMessageStore(EditContext);
+                    }
+
+                    _parsingValidationMessages.Add(FieldIdentifier, validationErrorMessage);
+
+                    // Since we're not writing to CurrentValue, we'll need to notify about modification from here
+                    EditContext.NotifyFieldChanged(FieldIdentifier);
+                }
+
+                // We can skip the validation notification if we were previously valid and still are
+                if (parsingFailed || _previousParsingAttemptFailed)
+                {
+                    EditContext.NotifyValidationStateChanged();
+                    _previousParsingAttemptFailed = parsingFailed;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Formats the value as a string. Derived classes can override this to determine the formating used for <see cref="CurrentValueAsString"/>.
+        /// </summary>
+        /// <param name="value">The value to format.</param>
+        /// <returns>A string representation of the value.</returns>
+        protected virtual string FormatValueAsString(T value)
+            => value?.ToString();
+
+        /// <summary>
+        /// Parses a string to create an instance of <typeparamref name="T"/>. Derived classes can override this to change how
+        /// <see cref="CurrentValueAsString"/> interprets incoming values.
+        /// </summary>
+        /// <param name="value">The string value to be parsed.</param>
+        /// <param name="result">An instance of <typeparamref name="T"/>.</param>
+        /// <param name="validationErrorMessage">If the value could not be parsed, provides a validation error message.</param>
+        /// <returns>True if the value could be parsed; otherwise false.</returns>
+        protected virtual bool TryParseValueFromString(string value, out T result, out string validationErrorMessage)
+            => throw new NotImplementedException($"Components that inherit from {nameof(InputBase<T>)} must override " +
+                $"{nameof(TryParseValueFromString)} in order to use {nameof(CurrentValueAsString)}.");
+
+        /// <summary>
+        /// Gets a string that indicates the status of the field being edited. This will include
+        /// some combination of "modified", "valid", or "invalid", depending on the status of the field.
+        /// </summary>
+        protected string FieldClass
```

Comment:
Is this a good name for the property? It doesn't correspond to the comment above.